logger/logger.go

```diff
@@ -34,17 +34,29 @@ import (
 	"github.com/op/go-logging"
 )
 
+// Channel specifies the logging channel. Can be one of CLI, API or LSP
+type channel int
+
 const (
 	logsDirectory    = "logs_directory"
 	logs             = "logs"
 	GaugeLogFileName = "gauge.log"
 	apiLogFileName   = "api.log"
 	lspLogFileName   = "lsp.log"
+	// CLI indicates gauge is used as a CLI.
+	CLI channel = iota
+	// API indicates gauge is in daemon mode. Used in IDEs.
+	API
+	// LSP indicates that gauge is acting as an LSP server.
+	LSP
 )
 
 var level logging.Level
-var isWindows bool
 var customLogger CustomLogger
+var activeLogger *logging.Logger
+var fileLogFormat = logging.MustStringFormatter("%{time:15:04:05.000} %{message}")
+var isLSP bool
+var initialized bool
 
 type CustomLogger interface {
```

Comment:
This interface can be removed now that lang package invokes logger through its own logger. The lang package can do any custom logging it needs on its own.